code/modules/antagonists/heretic/magic/moon_parade.dm

```diff
@@ -67,29 +68,30 @@
 		visible_message(span_warning("The parade hits [victim] and a sudden wave of clarity comes over you!"))
 		return PROJECTILE_DELETE_WITHOUT_HITTING
 
+	return ..()
 
 /obj/projectile/moon_parade/on_hit(atom/target, blocked = 0, pierce_hit)
 	. = ..()
+	if(. == BULLET_ACT_BLOCK || !isliving(target))
+		return
+
 	var/mob/living/victim = target
 
-	RegisterSignal(victim, COMSIG_MOB_CLIENT_PRE_LIVING_MOVE, PROC_REF(moon_block_move), override=TRUE)
+	if(!(victim in mobs_hit))
+		RegisterSignal(victim, COMSIG_MOB_CLIENT_PRE_LIVING_MOVE, PROC_REF(moon_block_move))
+		RegisterSignal(victim, COMSIG_QDELETING, PROC_REF(clear_mob))
+		victim.AddComponent(/datum/component/leash, src, distance = 1)
+		victim.balloon_alert(victim, "you feel unable to move away from the parade!")
```

Comment:
this balloon alert is way too long

Reply:
It is but I'm just moving code so I'm not touching it in this pr